plugins/scs/src/main/java/org/eclipse/xpanse/plugins/scs/enums/ScsTerraformResourceProperties.java

```diff
@@ -0,0 +1,230 @@
+    static class ScsVmProperties extends DeployResourceProperties {
+
+        @Override
+        public DeployResourceKind getResourceKind() {
+            return DeployResourceKind.VM;
+        }
+
+        @Override
+        public Map<String, String> getResourceProperties() {
+            Map<String, String> map = new HashMap<>();
+            map.put("ip", "access_ip_v4");
+            map.put("image_id", "image_id");
+            map.put("image_name", "image_name");
+            map.put("region", "region");
+            return map;
+        }
+    }
+
+    /**
+     * Scs volume properties.
+     */
+    static class ScsVolumeProperties extends DeployResourceProperties {
+        @Override
+        public DeployResourceKind getResourceKind() {
+            return DeployResourceKind.VOLUME;
+        }
+
+        @Override
+        public Map<String, String> getResourceProperties() {
+            Map<String, String> map = new HashMap<>();
+            map.put("size", "size");
+            map.put("type", "volume_type");
+            return map;
+        }
+    }
+
+    /**
+     * Scs publicIp properties.
+     */
+    static class ScsPublicIpProperties extends DeployResourceProperties {
+
+        @Override
+        public DeployResourceKind getResourceKind() {
+            return DeployResourceKind.PUBLIC_IP;
+        }
+
+        @Override
+        public Map<String, String> getResourceProperties() {
+            Map<String, String> map = new HashMap<>();
+            map.put("ip", "address");
+            return map;
+        }
+    }
+
+    /**
+     * Scs vpc properties.
+     */
+    static class ScsVpcProperties extends DeployResourceProperties {
+
+        @Override
+        public DeployResourceKind getResourceKind() {
+            return DeployResourceKind.VPC;
+        }
+
+        @Override
+        public Map<String, String> getResourceProperties() {
+            Map<String, String> map = new HashMap<>();
+            map.put("region", "region");
+            map.put("mtu", "mtu");
+            return map;
+        }
+    }
+
+    /**
+     * Scs subnet properties.
+     */
+    static class ScsSubnetProperties extends DeployResourceProperties {
+
+        @Override
+        public DeployResourceKind getResourceKind() {
+            return DeployResourceKind.SUBNET;
+        }
+
+        @Override
+        public Map<String, String> getResourceProperties() {
+            Map<String, String> map = new HashMap<>();
+            map.put("vpc", "network_id");
+            map.put("subnet", "cidr");
+            map.put("gateway", "gateway_ip");
+            return map;
+        }
+    }
+
+
+    /**
+     * Scs keypair properties.
+     */
+    static class ScsKeyPairProperties extends DeployResourceProperties {
+
+        @Override
+        public DeployResourceKind getResourceKind() {
+            return DeployResourceKind.KEYPAIR;
+        }
+
+        @Override
+        public Map<String, String> getResourceProperties() {
```

Comment:
in case of openstack and SCS, the public/private key has an error when we try to store it in DB.

![image](https://github.com/eclipse-xpanse/xpanse/assets/54129659/cefada04-f24b-4c23-b434-f98c2a6768cc)

Reply:
I updated the length of `p_value`.